npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -38,14 +38,16 @@ def self.yarn_version_numeric(yarn_lock)
 
       sig { params(pnpm_lock: DependencyFile).returns(Integer) }
       def self.pnpm_version_numeric(pnpm_lock)
-        if pnpm_lockfile_version(pnpm_lock).to_f >= 9.0
-          9
-        elsif pnpm_lockfile_version(pnpm_lock).to_f >= 6.0
-          8
-        elsif pnpm_lockfile_version(pnpm_lock).to_f >= 5.4
+        pnpm_version = pnpm_lockfile_version(pnpm_lock).to_f;
+
+        if pnpm_version < 5.4
+          6
+        elsif pnpm_version < 6
           7
+        elsif pnpm_version < 9
+          8
         else
-          6
+          pnpm_version.to_i
```

Comment:
Much better! :)

Reply:
OK, but that's the current state of things, right?  This is a better version of what is currently already "unsafe and probably wrong", right?

Reply:
What's unsafe or wrong with this exactly ?
```ruby
if pnpm_lockfile_version(pnpm_lock).to_f >= 9.0
  9
elsif pnpm_lockfile_version(pnpm_lock).to_f >= 6.0
  8
elsif pnpm_lockfile_version(pnpm_lock).to_f >= 5.4
  7
else
  6
end
```

Reply:
> What's unsafe or wrong with this exactly ?

- Given `pnpm_lockfile_version(pnpm_lock).to_f == 10`, then it will use version `9`
- Given `pnpm_lockfile_version(pnpm_lock).to_f == 42`, then it will use version `9`
- Given `pnpm_lockfile_version(pnpm_lock).to_f == 8576309`, then it will use version `9`

Invert the logic, and you can future proof it.  It makes no sense to constrain it the way it is.